rust-runtime/aws-smithy-runtime-api/src/client/http/request.rs

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::box_error::BoxError;
+use crate::client::interceptors::context::Error;
+use aws_smithy_http::body::SdkBody;
+use http as http0;
+use std::borrow::Cow;
+use std::fmt::{Debug, Formatter};
+use std::str::FromStr;
+
+/// An HTTP Request Type
+pub struct Request<B = SdkBody> {
+    inner: http0::Request<B>,
+}
+
+impl<B: Debug> Debug for Request<B> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self.inner)
+    }
+}
+
+impl<B> Request<B> {
+    /// Converts this request into an http 0.x request.
+    ///
+    /// Depending on the internal storage type, this operation may be free or it may have an internal
+    /// cost.
+    pub fn into_http0x(self) -> http::Request<B> {
```

Comment:
Hypothetically, if there was ever a http 0.3 release before 1.0, this would need an additional method that would break the naming convention, right?

Reply:
yeah, we should either make this a From impl or name it more precisely

Reply:
Should it be fallible in case it's not possible to convert in the future (due to losing information or something)?